context/app/static/js/components/savedLists/hooks.ts

```diff
@@ -0,0 +1,212 @@
+  useUkvApiURLs,
+  useFetchSavedEntitiesAndLists,
+  copySavedItemsToRemoteStore,
+  saveEntityRemote,
+  saveEntitiesRemote,
+  deleteEntityRemote,
+  deleteEntitiesRemote,
+  createListRemote,
+  addEntityToListRemote,
+  addEntitiesToListRemote,
+  removeEntityFromListRemote,
+  removeEntitiesFromListRemote,
+  deleteListRemote,
+  editListRemote,
+} from 'js/components/savedLists/api';
+
+const savedEntitiesSelector = (state: SavedEntitiesStore) => ({
+  savedLists: state.savedLists,
+  savedEntities: state.savedEntities,
+  listsToBeDeleted: state.listsToBeDeleted,
+  setEntities: state.setEntities,
+  setLists: state.setLists,
+  saveEntity: state.saveEntity,
+  saveEntities: state.saveEntities,
+  deleteEntity: state.deleteEntity,
+  deleteEntities: state.deleteEntities,
+  createList: state.createList,
+  addEntityToList: state.addEntityToList,
+  addEntitiesToList: state.addEntitiesToList,
+  removeEntityFromList: state.removeEntityFromList,
+  removeEntitiesFromList: state.removeEntitiesFromList,
+  queueListToBeDeleted: state.queueListToBeDeleted,
+  deleteQueuedLists: state.deleteQueuedLists,
+  deleteList: state.deleteList,
+  editList: state.editList,
+});
+
+/* Hook that wraps the savedEntities store and provides functions for saving, deleting, and managing lists. */
+function useSavedLists() {
+  const urls = useUkvApiURLs();
+  const { groupsToken, isAuthenticated } = useAppContext();
+  const { setTransferredToProfile } = useSavedListsAlertsStore();
+  const { savedEntities, savedLists, isFirstRemoteFetch, isLoading } = useFetchSavedEntitiesAndLists({
+    urls,
+    groupsToken,
+    isAuthenticated,
+  });
+
+  const localStore = useLocalSavedEntitiesStore(savedEntitiesSelector);
+  const remoteStore = useRemoteSavedEntitiesStore(savedEntitiesSelector);
+  const store = isAuthenticated ? remoteStore : localStore;
+
+  const storeHasBeenSetRef = useRef(false);
+  const wasLoadingRef = useRef(isLoading);
+
+  useEffect(() => {
+    if (!isAuthenticated || isLoading || !wasLoadingRef.current) {
+      return;
+    }
+
+    if (isFirstRemoteFetch) {
+      // If a user logs into their account for the first time since the My Lists update
+      // on a device that has saved entities and lists, we copy those to their remote store and show them an alert.
+      // This only happens once.
+      setTransferredToProfile(true);
+      const { savedEntities: savedEntitiesLocal, savedLists: savedListsLocal } = useLocalSavedEntitiesStore.getState();
+
+      copySavedItemsToRemoteStore({ savedEntitiesLocal, savedListsLocal, urls, groupsToken });
+      store.setEntities(savedEntitiesLocal);
+      store.setLists(savedListsLocal);
+    } else {
+      // If the user already has saved entities and lists in their remote store, we need to copy those over to the local store.
+      store.setEntities(savedEntities);
+      store.setLists(savedLists);
+    }
+    storeHasBeenSetRef.current = true;
+    wasLoadingRef.current = isLoading;
```

Comment:
Do we need both `storeHasBeenSetRef` and `wasLoadingRef`, I believe they will always have the same value?